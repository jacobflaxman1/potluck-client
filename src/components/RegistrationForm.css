.RegistrationPage {
    text-align: center;
}

.RegistrationForm {
    text-align: center;
}
.registration-input{
    margin: 5px;;
    width: 100%;
    max-width: 280px;
    border-radius: 3px;
    overflow: hidden;
    
    align-self: center;

}

.input-fields:focus{
    border: none;
}

label {

    font-size: 16px;
    
    font-weight: 500;

}