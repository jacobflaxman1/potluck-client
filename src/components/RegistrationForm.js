import React, { Component } from 'react'
import AuthApiService from '../services/auth-api-service'

export default class RegistrationForm extends Component {

    state = { error: null }

    handleSubmit = e => {
        e.preventDefault()
        const { user_name, nick_name, password } = e.target
        AuthApiService.postUser({
            user_name: user_name.value,
            nick_name: nick_name.value,
            password: password.value
        })
        .then(user => {
            user_name.value = ''
            nick_name.value = ''
            password.value = ''
            this.props.history.push('/login')
        })
        .catch(res => {
            this.setState({ error: res.error })
        })
    }

    render() {
        const { error } = this.state
        return(
            <form className = 'RegistrationForm' onSubmit = {this.handleSubmit}>
                <div> { error && <p>{error} </p> } </div>
                <label htmlFor = 'registrationForm-userName'> User Name </label>
                <input name = 'user_name' required type = 'text'/>
                
                <label htmlFor = 'registrationForm-nickName'> Nick Name </label>
                <input name = 'nick_name' required type = 'text'/>

                <label htmlFor = 'registrationForm-password'> Password </label>
                <input name = 'password' required type = 'text'/>

                <button type= 'submit'> Register </button>
            </form>
        )
    }
}