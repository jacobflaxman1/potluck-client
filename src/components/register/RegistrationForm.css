.RegistrationPage {
    text-align: center;
}

.RegistrationForm {
    display: flex;
    justify-content: space-evenly;
    align-content: space-around;
    flex-direction: column;
    max-width: 500px;
	padding: 10px 20px;
	background: #f4f7f8;
	margin: 10px auto;
	padding: 20px;
	background: #f4f7f8;
	border-radius: 8px;
	font-family: Georgia, "Times New Roman", Times, serif;
}
.registration-input{
    margin: 5px;
    width: 100%;
    max-width: 280px;
    border-radius: 3px;
    overflow: hidden;

}

.input-fields:focus{
    border: none;
}


.register-input {
    display: inline-flex;
    justify-content: space-evenly;
    align-self: center;
    flex-direction: column;

    border-radius: 6px;
    padding: 5px;
    margin: 10px;
}

.register {
    align-self: center;
    border: none;
    border-radius: 6px;
    color: #05386B;
    font-weight: 500;
    font-size: 16px;
}

.register:hover {
    background-color: rgb(59, 108, 156);
}