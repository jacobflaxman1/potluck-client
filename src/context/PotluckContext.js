import React, { Component } from 'react'
import itemApiService from '../services/item-api-service'
import UserApiService from '../services/user-api-service'

const PotluckContext = React.createContext({
    potluck: [],
    items: [],
    error:null,
    authToken: null,
    users: [],
    adminUser: [],
    usersInPotluck: [],
    setEror: () => {},
    setPotluck: () => {},
    setItems: () => {},
    clearError: () => {},
    clearPotluck: () => {},
    saveAuthToken: () => {},
    clearAuthToken: () => {},
    addPostedPotluck: () => {},
    updateItem: () => {},
    getAllUserNames: () => {},
    getGuestUsersInPotluck: () => {},
    getAdminUserInPotluck: () => {},
    // LIST CONTEXT
    potluckList: [],
    user: '',
    setPotluckList: () => {},
    setUser: () => {},
    setShowForm: () => {},
    showForm: false,
    setExpanded: () => {},
    render: false,
    setRender: () => {},
    clearPotluckInList: () => {}
})

export default PotluckContext

export class PotluckProvider extends Component {
    state = {
        potluck: [],
        items: [],
        users: [],
        adminUser: [],
        usersInPotluck:[],
        error: null,
        authToken: null,
        showForm: false,
        //LIST STATE
        potluckList: [],
        user: '',
        render: false
    }

    setRender = () => {
        this.setState({ render: !this.state.render})
    }

    setShowForm = () => {
        this.setState({ showForm: !this.state.showForm })
    }


    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setPotluck = (potluck) => {
        this.setState({ potluck })
    }

    setItems = (items) => {
        this.setState({ items })
    }

    clearPotluck = () => {
        this.setPotluck([])
    }

    clearPotluckInList = (id) => {
        let newList = [...this.state.potluckList]
        let index = this.state.potluckList.findIndex(el => el.potluck_id === id)

        if(index !== -1) {
            newList.splice(index, 1)
            this.setState({potluckList: newList})
        }
        
    }

    saveAuthToken = (authToken) => {
        this.setState({ authToken })
    }

    clearAuthToken = () => {
        this.setState({ authToken: null })
    }

    addPostedPotluck = (newPotluck) => {
        this.setState({ potluckList: [...this.state.potluckList, {...newPotluck, expanded: true}] })
    }
    
    updateItem = (id) => {
        itemApiService.updateItem(id)
            .then((data) => {
                const newItemData = this.state.items.map((item) => {
                    if(id === item.item_id) {
                        return data
                    }
                    return item
                })
                this.setState({items: newItemData})
        })
    }

    getAllUserNames = () => {
        UserApiService.getUsers()
            .then((data) => {
                let userNames = data.rows.map(user => {
                    return user
                })
                this.setState({ users: userNames})
            })
    }
    getGuestUsersInPotluck = (id) => {
        console.log(id)
        UserApiService.getUsersByPotluck(id)
            .then((data) => {
                let userNames = data.guests.map(user => {
                    return user
                })
                this.setState({ usersInPotluck: userNames})
        })
    }
    getAdminUserInPotluck = (id) => {
        UserApiService.getUsersByPotluck(id) 
            .then((data) => {
                let adminUser = data.admin.map(ad => {
                    return ad
                })
                this.setState({ adminUser: adminUser})
            })
    }
    //LIST FUNCTIONS
    setPotluckList = potluckList => {
        const expandedPotluckList = potluckList.map(data => {
            return {...data, expanded: false}
        })

        this.setState({ potluckList: expandedPotluckList })
      }

    setUser = (user) => {
        this.setState({ user })
      }


    setExpanded = (id) => {
        const index = this.state.potluckList.findIndex(el => el.potluck_id === id)
        let newPotluckState = [...this.state.potluckList]

        newPotluckState[index] = {...newPotluckState[index], expanded: !newPotluckState[index].expanded}
        this.setState({ potluckList: newPotluckState })
    }

    render() {
        const value = {
            potluck: this.state.potluck,
            items: this.state.items,
            users: this.state.users,
            usersInPotluck: this.state.usersInPotluck,
            adminUser: this.state.adminUser,
            error: this.state.error,
            authToken: this.state.authToken,
            potluckList: this.state.potluckList,
            user: this.state.user,
            setItems: this.setItems,
            setError: this.setError,
            setPotluck: this.setPotluck,
            clearPotluck: this.clearPotluck,
            saveAuthToken: this.saveAuthToken,
            clearAuthToken: this.clearAuthToken,
            addPostedPotluck: this.addPostedPotluck,
            updateItem: this.updateItem,
            getAllUserNames: this.getAllUserNames,
            getGuestUsersInPotluck: this.getGuestUsersInPotluck,
            getAdminUserInPotluck: this.getAdminUserInPotluck,
            setPotluckList: this.setPotluckList,
            setUser: this.setUser,
            clearError: this.clearError,
            showForm: this.state.showForm,
            setShowForm: this.setShowForm,
            setExpanded: this.setExpanded,
            render: this.state.render,
            setRender: this.setRender,
            clearPotluckInList: this.clearPotluckInList

        }
        return(
            <PotluckContext.Provider value = {value}>
                {this.props.children}
            </PotluckContext.Provider>
        )
    }

}