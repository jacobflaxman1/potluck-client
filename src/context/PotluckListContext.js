import React, { Component } from 'react'

const PotluckListContext = React.createContext({
  potluckList: [],
  error: null,
  user: '',
  setError: () => {},
  clearError: () => {},
  setPotluckList: () => {},
  setUser: () => {}
})
export default PotluckListContext

export class PotluckListProvider extends Component {
  state = {
    potluckList: [],
    user: '',
    error: null,
  };

  setPotluckList = potluckList => {
    this.setState({ potluckList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  setUser = (user) => {
    this.setState({ user })
  }

  render() {
    const value = {
      potluckList: this.state.potluckList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setPotluckList: this.setPotluckList,
      setUser: this.setUser
    }
    return (
      <PotluckListContext.Provider value={value}>
        {this.props.children}
      </PotluckListContext.Provider>
    )
  }
}
