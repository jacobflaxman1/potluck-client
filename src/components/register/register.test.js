import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { PotluckProvider } from '../../context/PotluckContext'
import RegistrationForm from './RegistrationForm';


describe(`Adds all of the potluck views to dom without crashing`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(
        <PotluckProvider>
            <RegistrationForm />
        </PotluckProvider>
    )
    .find('.RegistrationForm')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})