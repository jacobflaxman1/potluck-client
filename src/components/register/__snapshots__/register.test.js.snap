// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds all of the potluck views to dom without crashing renders without crashing 1`] = `null`;
